src/main/java/io/ticktok/rest/RestTicktokClient.java

```diff
@@ -1,67 +1,49 @@
 package io.ticktok.rest;
 
 import com.google.gson.Gson;
-import io.ticktok.TicktokApi;
+import io.ticktok.Ticktok;
 import io.ticktok.TicktokOptions;
 import io.ticktok.register.Clock;
 import io.ticktok.register.RegisterClockRequest;
-import org.apache.http.HttpHeaders;
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.entity.StringEntity;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.message.BasicHeader;
-import org.apache.http.util.EntityUtils;
+import io.ticktok.validator.ClockRequest;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
 
 import java.io.IOException;
 
+import static io.ticktok.TicktokApi.*;
+
 public class RestTicktokClient {
     private final String token;
-    private HttpClient client;
     private final String domain;
 
     public RestTicktokClient(TicktokOptions options){
         this.domain = options.getDomain();
         this.token = options.getToken();
-        this.client = HttpClientBuilder.create().build();
-    }
-
-    public String post(String entry, String entryDomain) throws IOException {
-        return handleBodyCall(new HttpPost(this.domain + "/" + entryDomain), entry);
-    }
-
-    private String handleBodyCall(HttpEntityEnclosingRequestBase request, String entry) throws IOException {
-        request.setEntity(new StringEntity(entry));
-        request.setHeaders(defaultHeaders());
-        HttpResponse response = client.execute(request);
-        String restCallResponse = EntityUtils.toString(response.getEntity());
-        request.releaseConnection();
-        validateResponseWasOk(response);
-        return restCallResponse;
     }
 
-    private BasicHeader[] defaultHeaders() {
-        return new BasicHeader[]{
-                new BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json"),
-                new BasicHeader(HttpHeaders.ACCEPT, "application/json"),
-                new BasicHeader(HttpHeaders.AUTHORIZATION, this.token)
-        };
+    public Clock register(ClockRequest clockRequest) {
+        return new Gson().fromJson(call(clockRequest.getName(), clockRequest.getSchedule()), Clock.class);
```

Comment:
what happen if you get 400? you should address it since the schedule can be invalid